### PR TITLE
Fix toplevel transpile

### PR DESCRIPTION
**Pull request by aethos**
makes testing slow but makes the toplevel do the right stuff

now toplevel will compile your file for you and generate an executable
eg: ./oscar -c divide.oscar -o divide_exec
will generate the cpp file divide.cpp and executabledivide_exec

also added an error for scanners

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,8 +2,8 @@ TARGET=oscar.native
 OSCAR=oscar
 STDLIB=stdlib.oscar
 NATIVE=native
-ACTOR=$(NATIVE)/actor
-IMMUT=$(NATIVE)/immut
+ACTOR=NATIVE/actor
+IMMUT=NATIVE/immut
```
Comment (Howon) [inline Makefile:6]: Don't think this will work

Comment (Howon) [reply Makefile:6]: why did you override this?

File: Makefile [hunk 2]
```diff
@@ -28,6 +30,16 @@ oscar :
 	cp ./src/$(IMMUT)/* ./$(BUILDDIR) && \
 	echo 'Oscar Compiler Succesfully Built'
 
+library : clean_library
+	@echo 'Building Oscar Library'
+	@mkdir /usr/local/include/oscar && \
+	cp src/native/immut/collection.hpp  /usr/local/include/oscar/ && \
+	cp src/native/immut/immut.hpp 	   /usr/local/include/oscar/ && \
+	cp src/native/immut/list.hpp		   /usr/local/include/oscar/ && \
```
Comment (Howon) [inline Makefile:38]: can we fix indentation

File: Makefile [hunk 3]
```diff
@@ -43,15 +55,36 @@ parser.ml parser.mli : parser.mly
...
-	echo Oscar build cleaned
+	echo 'Oscar Build Cleaned'
+
+clean_library :
+	@echo 'Cleaning Oscar Library'
+	@rm -rf /usr/local/include/oscar && \
+	echo 'Oscar Library Cleaned'
+    
+
+# Generated by ocamldep *.ml *.mli
```
Comment (Howon) [inline Makefile:77]: why include this pile of garbage code?

File: src/oscar.ml [hunk 4]
```diff
@@ -6,47 +6,56 @@ open Sast
...
+    (fprintf stderr "Parser: Syntax error at line %d, char %d - %d: \"%s\" \n"
     s.pos_lnum (s.pos_cnum - s.pos_bol)
     (f.pos_cnum - f.pos_bol) (Lexing.lexeme lexbuf))
 
 let _ =
   let arg_len = Array.length Sys.argv in
   let (action, oscar) =
-    (if (arg_len = 1 || arg_len > 3) then
-      let _ = print_endline("Usage: ./oscar [-l|-c|-s] *.oscar") in
+    (if (arg_len = 1 || arg_len = 4 || arg_len > 5) then
```
Comment (Howon) [inline src/oscar.ml:36]: arg_len = 4 || arg_len > 5
srsly

**Comment by Howon** [comment]
solution to using git is not to override everything in remote with
whatever incorrect version you have in your local repo

File: Makefile (hunk 1 above)
Comment (aethos) [reply Makefile:6]: got it from master

File: Makefile (hunk 2 above)
Comment (aethos) [reply Makefile:38]: yes

File: Makefile (hunk 3 above)
Comment (aethos) [reply Makefile:77]: removed

File: src/oscar.ml (hunk 4 above)
Comment (aethos) [reply src/oscar.ml:36]: this is necessary -- acceptable arg lengths are only 3 or 5. I've changed it to reflect that
we can do != but for some reason that's deprecated

**Pull request closed without merging**